### ISSUE TITLE
Add system management agent to snap

### ISSUE THREAD

**Issue by anonymouse64** [edgexfoundry/edgex-go#624]
For Delhi, there will be an additional service, the Micro Service Management. We need to add this into the snap.

See also issue #581

**Comment by anonymouse64** [comment]
@akramtexas I'm having some difficulty starting up the system management agent from the snap and am hoping you can help me diagnose the issue here. When I run the system management agent from the snap , it panics:

```
+ /snap/edgexfoundry/x1/bin/sys-mgmt-agent --consul
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x658e63]

goroutine 33 [running]:
github.com/edgexfoundry/edgex-go/vendor/github.com/hashicorp/consul/api.(*Client).newRequest(0x0, 0x7aadfc, 0x3, 0x7aef68, 0x12, 0x0)
	/home/ijohnson/git/edgex-go/parts/edgex-go/build/.gopath/src/github.com/edgexfoundry/edgex-go/vendor/github.com/hashicorp/consul/api/api.go:644 +0x63
github.com/edgexfoundry/edgex-go/vendor/github.com/hashicorp/consul/api.(*Agent).Services(0xc42004cd80, 0x0, 0x0, 0x0)
	/home/ijohnson/git/edgex-go/parts/edgex-go/build/.gopath/src/github.com/edgexfoundry/edgex-go/vendor/github.com/hashicorp/consul/api/agent.go:245 +0x6d
github.com/edgexfoundry/edgex-go/internal/pkg/consul.GetServiceEndpoint(0x7b04cd, 0x15, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/home/ijohnson/git/edgex-go/parts/edgex-go/build/.gopath/src/github.com/edgexfoundry/edgex-go/internal/pkg/consul/client.go:108 +0x76
github.com/edgexfoundry/edgex-go/internal/pkg/startup.Endpoint.Monitor(0x7b04cd, 0x15, 0x7aac25, 0x1, 0x1, 0xc42031a5e0, 0x16, 0x3a98, 0xc420270ae0)
	/home/ijohnson/git/edgex-go/parts/edgex-go/build/.gopath/src/github.com/edgexfoundry/edgex-go/internal/pkg/startup/endpoint.go:29 +0x1c8
created by github.com/edgexfoundry/edgex-go/pkg/clients/general.(*generalRestClient).init
	/home/ijohnson/git/edgex-go/parts/edgex-go/build/.gopath/src/github.com/edgexfoundry/edgex-go/pkg/clients/general/client.go:42 +0xd5
```

We are running it like this:
```bash
cd $SNAP_DATA/config/sys-mgmt-agent
$SNAP/bin/sys-mgmt-agent --consul &
```
and the configuration file is located in `$SNAP_DATA/config/sys-mgmt-agent/res/configuration.toml` and looks like this:

```toml
ApplicationName = 'edgex-sys-mgmt-agent'
ConsulProfilesActive = 'go'
ValidateCheck = false
AppOpenMsg = 'This is the System Management Agent Service'
FormatSpecifier = '%(\\d+\\$)?([-#+ 0(\\<]*)?(\\d+)?(\\.\\d+)?([tT])?([a-zA-Z%])'
ServicePort = 48090
ServiceTimeout = 5000
ServiceAddress = 'localhost'
CheckInterval = '10s'
EnableRemoteLogging = false
LoggingFile = '/var/snap/edgexfoundry/common/sys-mgmt-agent.log'

[Clients]
  [Clients.Notifications]
  Protocol = 'http'
  Host = 'localhost'
  Port = 48060

  [Clients.Command]
  Protocol = 'http'
  Host = 'localhost'
  Port = 48082

  [Clients.Metadata]
  Protocol = 'http'
  Host = 'localhost'
  Port = 48081

  [Clients.Logging]
  Protocol = 'http'
  Host = 'localhost'
  Port = 48061

  [Clients.CoreData]
  Protocol = 'http'
  Host = 'localhost'
  Port = 48080

  [Clients.Scheduler]
  Protocol = 'http'
  Host = 'localhost'
  Port = 48085

  [Clients.Export]
  Protocol = 'http'
  Host = 'localhost'
  Port = 48071

  [Clients.Distro]
  Protocol = 'http'
  Host = 'localhost'
  Port = 48070
```

**Comment by akramtexas** [comment]
@anonymouse64 The details you've provided are detailed and helpful. I'll be in touch just as soon as I've established line of sight into the root cause of the issue. Copying @jpwhitemn, whose feedback and insight have been instrumental in helping me in this work, to keep in the loop. Of course, and needles to say, I'm responsible for any issue(s)--such as the one you've pointed out here--and will address them, thanks 🐞

**Comment by akramtexas** [comment]
@anonymouse64 After investigation, narrowed down the area where the bug might be lurking (to two areas)— (1) by examining the stack trace, it appears that a call is being made to Consul (which is either not responding or is down, and hence, possibly, the Go-runtime-emitted panic) and / or (2) currently we don't have a manifest.toml for Docker to consume, which can't be helping 🐛 So I have a fix that I'm going to shortly be testing locally @jpwhitemn.

**Comment by anonymouse64** [comment]
@akramtexas I should have mentioned this, but Consul is indeed running when this panic happens.

**Comment by akramtexas** [comment]
@anonymouse64 Gotcha, thanks for clarifying. Every bit of data helps (narrow down where the bug lurks!), thanks.

**Comment by anonymouse64** [comment]
@akramtexas could this issue have anything to do with issue #715 ?

**Comment by akramtexas** [comment]
@anonymouse64 Your comment above, the question, actually, slipped my radar, sorry. Just noticed it: I do not see it having any connection with the scenario in this issue...

**Comment by akramtexas** [comment]
@anonymouse64 The fix, which is in the branch per the following PR, is ready for your testing, too, myself having tested it recently: [PR #727](https://github.com/edgexfoundry/edgex-go/pull/727) @tsconn23 @jpwhitemn - Thanks, everyone 👍

**Comment by anonymouse64** [comment]
This was merged to master for Delhi with #951
